src/main/kotlin/org/rust/cargo/toolchain/CargoMessage.kt

```diff
@@ -0,0 +1,77 @@
+package org.rust.cargo.toolchain
+
+import com.google.gson.Gson
+import com.google.gson.JsonArray
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+
+data class CargoTopMessage(
+    val message: CargoMessage,
+    val package_id: String,
+    val reason: String,
+    val target: Target
+) {
+    companion object {
+        fun fromJson(json: JsonObject): CargoTopMessage? {
+            if (json.getAsJsonPrimitive("reason").asString != "compiler-message") {
+                return null
+            }
+
+            return Gson().fromJson(json, CargoTopMessage::class.java)
+        }
+    }
+}
+
+data class CargoMessage(
+    val children: List<CargoMessage>,
+    val code: Code?,
+    val level: String,
+    val message: String,
+    val rendered: String?,
+    val spans: List<CargoSpan>
+)
+
+data class Code(
+    val code: String,
+    val explanation: String?)
+{
+    fun formatAsLink() : String {
+        val rustErrorIndexPrefix = "https://doc.rust-lang.org/error-index.html"
+
+        return "<a href=\"$rustErrorIndexPrefix#$code\">$code</a>"
```

Comment:
Actually, this `formatAsLink()` is unused and can be removed. But if you decide to add links to error codes (which would be very useful), it'll be nice to move the url or the whole error link format string to `RsConstants` and use it here and in `RsExplainFilter` that also contains this constant.